### ISSUE TITLE
No gas usage since 28th March

### ISSUE THREAD

**Issue by mcderd-coder** [HandyHat/ha-hildebrandglow-dcc#324]
I have no gas usage showing since 28th March, the previous week it shows usage for every day.

I am not sure if an HA update has broken it.

Still see the data in the Bright app

**Comment by barans99** [comment]
Same here. Real shame. Let me know if you find a fix.

I tried reinstall the integration completely, but no luck.

**Comment by mcderd-coder** [comment]
I will let you know if I find anything to fix it.

**Comment by Mikescotland** [comment]
No gas usage since a few days. Electricity stuck on first reading 10hrs before every day and never updated again. Bright app fine. No errors with debug log.

**Comment by bsimmo** [comment]
Same here and probably most users, seem to happen regularly. I wonder if it is a rate limiting thing, someone has to pay for their serer API usage.  More users, more HA integration hits.

Have you asked Hildebrandglow, they respond to email and seem very friendly and helpful when I suggested something to them.

**Comment by mcderd-coder** [comment]
Yes, I have contacted Hildebrandglow, no response yet, but rate limiting could be the cause.

**Comment by godfreym29** [comment]
My gas consumption has been missing for ages, yet it's visible using the Bright app on my phone. Something in the HA intergration is broken.

**Comment by mcderd-coder** [comment]
I wonder how we can get it fixed?

**Comment by JoeKlimis** [comment]
I am having the same issue, no Gas  usage information, 
getting the following errors in the log 

I hope this helps
Joe

Cannot connect: HTTPSConnectionPool(host='api.glowmarkt.com', port=443): Max retries exceeded with url: /api/v0-1/resource/84e5a66c-7041-42a2-b0db-6ce660442c95/catchup (Caused by NewConnectionError('<urllib3.connection.HTTPSConnection object at 0x7f7e7934bcd0>: Failed to establish a new connection: [Errno -3] Try again'))
11:33:28 AM – (ERROR) Hildebrand Glow (DCC) (custom integration) - message first occurred at 11:33:23 AM and shows up 2 times
Cannot connect: ('Connection aborted.', RemoteDisconnected('Remote end closed connection without response'))
11:33:28 AM – (ERROR) Hildebrand Glow (DCC) (custom integration) - message first occurred at 10:34:18 AM and shows up 3 times
Cannot connect: HTTPSConnectionPool(host='api.glowmarkt.com', port=443): Max retries exceeded with url: /api/v0-1/resource/eff6cee6-9b34-4e99-a576-e7c3eefe988e/catchup (Caused by NewConnectionError('<urllib3.connection.HTTPSConnection object at 0x7f7e48333a00>: Failed to establish a new connection: [Errno -3] Try again'))
11:33:28 AM – (ERROR) Hildebrand Glow (DCC) (custom integration) - message first occurred at 11:33:23 AM and shows up 4 times

**Comment by mcderd-coder** [comment]
UPDATE: I got a Gas reading today it shows todays consumption as -802.7 kWh yes that's negative 802.7 kWh 

So something is happening, but obviously, the readings are incorrect.

**Comment by eknirk** [comment]
See #330. You are probably not using any gas between midnight and 1am and that is what is reported for the whole day.

**Comment by abjwsm** [comment]
Hi I also have the same issue. Several weeks now. I have Electric but no gas data in both the phone app and Home assistant. It was working and I have changed nothing. I contacted Hildebrand & GlowMarket and they say they can see my Gas data at their end. I am using British Gas SMETS2 but the display unit has no built in WiFi. Are there any other methods available? Al

**Comment by mcderd-coder** [comment]
Hi, I am quite new to this, how do I apply the fix? Do I have to update something?

**Comment by e10kstarfire** [comment]
I think it is this https://github.com/HandyHat/ha-hildebrandglow-dcc/issues/330#issuecomment-1514780058

**Comment by mcderd-coder** [comment]
Although this was working again from when I updated sensor.py, it's now stopped again, I have no Gas data since 5th May, has it stopped again for others?

**Comment by e10kstarfire** [comment]
Not working for me with regards to Gas

**Comment by cannfoddr** [comment]
I've done a bit of a hack job and forked the base code and done a really simple merge of all the outstanding PRs.  I then took the resulting code folder and dropped it over my existing hacs install.   So far it seems to be working better.  I am seeing GAS and ELECTRIC entries with actual values.

However the energy dashboard is not showing figured for GAS - the entities have values but they dont appear on the energy dashboard.  Electricity is there however

**Comment by richard-scott** [comment]
@cannfoddr Do the entities have the correct attributes for the Dashboard to understand the value measurements e.g. £, kWh etc.

**Comment by cannfoddr** [comment]
I am not sure I think they are showing as KWH which seems electrical to me!

**Comment by cannfoddr** [comment]
Ok - I read another thread that's seems to suggest that GAS uses a Different DeviceClass to ELEC.  It looks like th integration is using 'energy' for both'. I just went into developer tools and changed the device class for my Gas usage from 'energy' to 'gas' and suddenly I have some gas usage in the graph!

I am looking at the integration code to see how the entity is getting created and how I can incorporate this change

**Comment by bsimmo** [comment]
> I am not sure I think they are showing as KWH which seems electrical to me!

kWh is a standard unit for Gas (it is just energy) you need to report it in whatever the API is reporting, let HA do any conversation.
m³ is the other one and a conversation isn't actually strightforward, but can be approximated. Mainly as there is a variable that changes in the conversion.

**Comment by bsimmo** [comment]
Not sure why you are messing with Gas.
Both come through and get logged and shown as standard, just sparodically, using some patch someone added.

I'm guessing the maintainer has abandoned this as it been out since the start of the year, and probably hits API limits near the end of the month, or used to. 
Maybe ask to take over?
